src/borda_clj/client.clj

```diff
@@ -0,0 +1,60 @@
+(ns borda-clj.client
+  (:use [slingshot.slingshot :only [throw+ try+]])
+  (:require [aleph.http :as http]
+            [byte-streams :as bs]
+            [cheshire.core :as json]))
+
+(defn collect
+  "Adds the given measurement (values and dimensions) to the given hashmap and caps the size of the hashmap at the given max-buffer-size"
+  [measurements max-buffer-size dimensions values]
+  (let [key (merge (sorted-map) dimensions)]
+    (if (contains? measurements key)
+      ; TODO: support stuff other than SUM here (e.g. AVG, MIN, MAX, etc.)
+      (assoc measurements key (merge-with + (get measurements key) values)) ; merge with existing in buffer
```

Comment:
There's [update](https://clojuredocs.org/clojure.core/update) so you don't need to explicitly get the key you want to update, so this should work:

```clojure
(update measurements key merge-with + values)
```

Reply:
Will do.